### ISSUE TITLE
[Bug]  OpenSSL error in every page except for start page

### ISSUE THREAD

**Issue by zehkaFlokinet** [iv-org/invidious#2753]
Note from @SamantazFox:

TL;DR: The BoringSSL library we use for QUIC interferes with the OpenSSL library installed on the system on RHEL-based distros. You should compile without QUIC using one of the commands below:

```bash
# Using the makefile
make DISABLE_QUIC=1

# Manually running the crystal build command
crystal build src/invidious.cr -Ddisable_quic [other build flags]
```

---

**Describe the bug**

I set up invidious on AlmaLinux using the updater script (with a small modification so it matches AlmaLinux too where ever centos is matched because they are compatible).
Setup works as intended but on the site and in the log ssl appear and no videos are shown from youtube.

**Steps to Reproduce**
Install invidious using the updater script and nginx reverse proxy on almalinux

**Logs**
```[error] UpdateDecryptFunctionJob : SSL_connect: error:1000007d:SSL routines:OPENSSL_internal:CERTIFICATE_VERIFY_FAILED```

**Screenshots**

**Additional context**
Error message (the page instructs me to include it here)

Title: `SSL_connect: error:1000007d:SSL routines:OPENSSL_internal:CERTIFICATE_VERIFY_FAILED (OpenSSL::SSL::Error)`
Date: `2022-01-01T22:55:15Z`
Route: `/feed/trending`
Version: `2021.12.28-a779cdd4 @ master`
<details>
<summary>Backtrace</summary>
<p>
   
```
SSL_connect: error:1000007d:SSL routines:OPENSSL_internal:CERTIFICATE_VERIFY_FAILED (OpenSSL::SSL::Error)
  from lib/lsquic/src/lsquic/patch.cr:174:11 in 'new:context:sync_close:hostname'
  from src/invidious/helpers/crystal_class_overrides.cr:34:5 in 'io'
  from /usr/share/crystal/src/http/client.cr:671:5 in 'send_request'
  from /usr/share/crystal/src/http/client.cr:603:18 in 'exec'
  from /usr/share/crystal/src/http/client.cr:713:5 in '_post_json'
  from src/invidious/trending.cr:19:14 in 'trending'
  from /usr/share/crystal/src/primitives.cr:266:3 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from lib/kemal/src/kemal/filter_handler.cr:21:7 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from src/invidious/helpers/static_file_handler.cr:167:15 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from lib/kemal/src/kemal/init_handler.cr:12:7 in 'process'
  from /usr/share/crystal/src/http/server.cr:500:5 in '->'
  from /usr/share/crystal/src/primitives.cr:266:3 in 'run'
  from ???
```
</p>
</details>

**Comment by unixfox** [comment]
Can you `curl https://www.youtube.com` and tell us if it works?

**Comment by zehkaFlokinet** [comment]
I forgot to include that into the bug report.
- curl works well with all sites tested including youtube.
- Time and date are correct so no certificate gets marked as outdated when it shouldn't be.
- Selinux is disabled to make sure that doesn't get in the way either
- I tested it with firewalld disabled in case it is a firewall problem, no success.

**Comment by unixfox** [comment]
Did you try to manually install Invidious using our guide: https://docs.invidious.io/Installation.md?

**Comment by zehkaFlokinet** [comment]
Yes that was the first thing i tried but i ran into the same issue. After that i reinstalled the server and started over fresh with the installer script because i believed it to be a mistake i made. Unfortunately i don't have logs from back then anymore

**Comment by unixfox** [comment]
How did you install CentOS? With these instructions: https://crystal-lang.org/install/on_centos/?

**Comment by zehkaFlokinet** [comment]
When i installed it manually i followed the tarball install from the invidious-guide, when i used the installer script it installed the RHEL 7 repo

**Comment by zehkaFlokinet** [comment]
I realized that RHEL 7 is not the ideal version for Almalinux 8.5 so i installed Crystal for CentOS 8 and recompiled invidious but the error still persists

**Comment by unixfox** [comment]
> When i installed it manually i followed the tarball install from the invidious-guide, when i used the installer script it installed the RHEL 7 repo

I don't understand exactly what you have tried, is installing crystal from https://crystal-lang.org/install/on_centos/ not working for you?
By the way we are on matrix and IRC if you need a faster help:
- matrix: https://matrix.to/#/#invidious:matrix.org
- IRC: https://web.libera.chat/?channel=#invidious

**Comment by unixfox** [comment]
> When i installed it manually i followed the tarball install from the invidious-guide, when i used the installer script it installed the RHEL 7 repo

I don't understand exactly what you have tried, is installing crystal from https://crystal-lang.org/install/on_centos/ not working for you?
By the way we are on matrix and IRC if you need a faster help:
- matrix: https://matrix.to/#/#invidious:matrix.org
- IRC: https://web.libera.chat/?channel=#invidious

**Comment by zehkaFlokinet** [comment]
I tried three ways of installing crystal.
First, during my manual attempt i used the crystal tarball.
Then i used the installer script, that installed the version for RHEL 7 for me.
And now i changed the repo the way https://crystal-lang.org/install/on_centos/ guides and reinstalled crystal with that.
But it didn't matter with which crystal version i compiled the result stayed the same.

If you prefer to use matrix we can switch there.

**Comment by SamantazFox** [comment]
So, I tried to reproduce (thanks @TheFrenchGhosty for the ArmaLinux VPS), and indeed, I have the same problem. I have no clue about the cause yet, so I'll continue investigation later today.

**Comment by TheFrenchGhosty** [comment]
@SamantazFox 
>I have no clue about the cause yet

Just a random guess: the openssl certificates aren't at the same place as other distro (`/etc/ssl/certs/`)

**Comment by SamantazFox** [comment]
> Just a random guess: the openssl certificates aren't at the same place as other distro (`/etc/ssl/certs/`)

There is a symlink for that, so it's a non-issue. See my edit above.

**Comment by TheFrenchGhosty** [comment]
@SamantazFox That's good to know, I have no idea what's the issue then

**Comment by zehkaFlokinet** [comment]
I played a bit with a test program:
```
require "http/client"

client = HTTP::Client.new("www.youtube.com", tls: true)
response = client.get "/"
puts response.status_code
```
Surprisingly that works without any problems (with crystal run as well as crystal build and then executing the binary)

When i compared the stracktraces of both binaries I noticed that the test binary opens /etc/pki/tls/cert.pem while invidious opens /etc/ssl/cert.pem, the second does not exist on almalinux.

Adding a symlink with ```ln -s /etc/pki/tls/cert.pem /etc/ssl/``` helps as a workaround but i consider it hacky

**Comment by SamantazFox** [comment]
So, after much testing, I figured out what's causing it! As I suspected, it was lsquic, but because I made a typo in the `Makefile`, the command I gave to compile without that QUIC library simply did nothing.

**Comment by TheFrenchGhosty** [comment]
@SamantazFox Issue is solved then?

(Maybe open a new one for lsquic on RHEL-based distro)

**Comment by SamantazFox** [comment]
The issue in itself is not solved. While it doesn't cause issues on other distros, it's still here and we need to find a way to avoid it.
